techSpec.js

```diff
@@ -0,0 +1,209 @@
+/*
+2. Introduction
+
+a. Overview, Problem Description, Summary, or Abstract
+
+    Context: Cat API is an Express.js application run on Node.js runtime. 
+
+    Problem: The Cat API currently does not have access to a cat database that persists. 
+    The information is generated when the API server is started. 
+    For this reason, for example, the ids are different every time around which made writing test cases trickier. 
+
+    Suggested solution: Implement a SQLite database which allows information to be persisted. 
+
+b. Glossary  or Terminology
+
+    SQL - structured query language 
+    SQLite - a file-based database engine that uses SQL syntax
+    sqlite3 - node module enables connection to SQLite databases in memory or disk file 
+
+    sqlite3 methods:
+        db.all - allows us to access all the rows in the dataset - this might offer the best flexibility for now
+        db.each - executes code for each row in a dataset
+        db.get - executes code for the FIRST row - use when we know only one row will return. 
+
+d. Goals or Product and Technical Requirements
+
+    Product requirements: 
+        1. We should have a pre-populated SQLite database on the file system. 
+        2. Upon start-up of the Cat API Express server, it should establish a connection to the SQL database - or should it?  
```

Comment:
Probably. Opening a database connection is usually [quite expensive](https://en.wikipedia.org/wiki/Database_connection#Pooling) in terms of performance, so it's an operation you'd want to minimise. It looks like the node module just keeps the connection open for you by default but that's something you could confirm yourself.

Reply:
Found a really good tutorial: https://stackabuse.com/a-sqlite-tutorial-with-node-js/
which as you suggested, opens the connection on server start-up and leaves it open. However it doesn't close it. 
Will continue researching tomorrow but haven't found anything definitive yet on Google about closing it (or when, to close it?!, since the server only closes by ctrl-c in Terminal)

Reply:
More research sounds good! You might be able to use a solution like this where the database connection is closed in a graceful shutdown scenario: https://flaviocopes.com/node-terminate-program/